core/server/src/main/java/alluxio/master/file/meta/InodeLockList.java

```diff
@@ -33,9 +35,27 @@
    * Locks the given inode in read mode, and adds it to this lock list.
    *
    * @param inode the inode to lock
+   * @param parent the expected parent inode
+   * @throws InvalidPathException if the inode is no long consistent with the caller's expectations
```

Comment:
it might make sense to be more explicit about what the caller expectations are between the different versions of the overloaded method